src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.Patch.cs

```diff
@@ -185,6 +185,10 @@ public abstract partial class InMemoryDocumentSessionOperations
             var pathScript = path.CompileToJavascript(_javascriptCompilationOptions.Value);
 
             var valueToUse = AddTypeNameToValueIfNeeded(path.Body.Type, value);
+            if (DocumentStore.Conventions.SaveEnumsAsIntegers && value is Enum)
```

Comment:
I can absolutley make that change, but can you please reconsider it once more and mayby have an internal discussion about it?

The documentation says that it changes the behaviour for storing and querying, but how can it be that patching is not part of that? I don't think that anybody feels that patching is not the same as storing values.

Also there is a bug that's not even related to `SaveEnumsAsIntegers`. 
`PatchOnDictionaryWithEnumKeyShouldWork` fails with the value `{"Checks":{"Engine":"Good","Gears":"Good","42":"Bad"}}` instead of the expected (`Checks":{"Engine":"Bad","Gears":"Good"}}`)and how is that supposed to be fixed by `SaveEnumsAsIntegersForPatching`? Since the user haven't specified `SaveEnumsAsIntegers`.

Should that bug fix have its own issue and PR?

Reply:
Just to make it clear - we'd like to accept your change as is. It is a good change. Although we're worried about backward compatibility. With the usage of a new, dedicated convention we will be guaranteed that we won't break anything for someone.

@aviv86 @garayx @ppekrol Your thoughts on this?

Reply:
I am bit worried that we need to maintain 2 conventions for basically same thing in the future, but it seems like the right decision for now, since we won't break code for customers.

Reply:
_As a user_ it makes no sense at all to have two properties that must both be set to true or false, how would you even document that?

Reply:
I wonder how this is handled in the other client libraries? Maybe you could motivate it as a potential "breaking" bug fix if all the other libraries handles it correctly?

Reply:
@arekpalinski & @garayx I have pushed a change that added `SaveEnumsAsIntegersForPatching` please take a close look at the tests comparing both `storedJson` and `patchJson` to understand the implications.